### PR TITLE
Flask login

### PR DESCRIPTION
**Pull request by jacobtizel**
This branch enhances the login functionality and provides easy access to the logged in user through current_user. This can be used as specified in the flask-login documentation to do things like checked attributes of the logged in user (ex/ current_user.username) or if they are logged in at all, through current_user.is_authenticated.

The registration and login pages are accessible through /login and /register. 

The @login_required decorator specifies a page as only accessible when logged in, and will redirect to the login page otherwise.

### CODE REVIEW

**Comment by jacobtizel** [comment]
Also, flash is used to make a popup for general errors or information for the user. These are only viewable on pages that extend  base.html, and they will pile up if not viewed like this.

**Comment by jacobtizel** [comment]
Updating via the requirements.txt is necessary, and you might want to drop your database if the password field isn't long enough (it's hashed now and a lot longer)

**Review by Graves067 — APPROVED** [review]
Looks good, changes to required code are reflected in the edits. Overall I do not believe there are changes that need to be made from my understanding. good work!

File: project/app.py [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
+app.config.from_object(Config)
+
+login_manager = LoginManager()
+login_manager.init_app(app)
+login_manager.login_view = "login"
+
 
 # Configuration for SQLAlchemy
-app.config['SQLALCHEMY_DATABASE_URI'] = 'mysql+pymysql://mango:COSC310=mcpw@127.0.0.1/mangochat'
+# app.config['SQLALCHEMY_DATABASE_URI'] = 'mysql+pymysql://mango:COSC310=mcpw@127.0.0.1/mangochat'
```
Comment (chrishill93) [inline project/app.py:18]: Is there a reason all of this is commented out?

Comment (jacobtizel) [reply project/app.py:18]: I moved the config to another file and left that there as an artifact. I can remove it.

Comment (chrishill93) [reply project/app.py]: Okay cool, that makes sense.  No need to remove it.

Comment (jacobtizel) [reply project/app.py]: In the name of refactoring, I have removed it.

**Comment by chrishill93** [comment]
I ran through registration once and it worked, ending on the success page

**Review by chrishill93 — APPROVED** [review]

**Pull request merged**